### PR TITLE
Document S3 Artifact type

### PR DESCRIPTION
**Pull request by benjaminws**
Light documentation to support https://github.com/spinnaker/spinnaker/issues/2595

### CODE REVIEW

File: reference/artifacts/index.md [hunk 1]
```diff
@@ -107,3 +107,14 @@ within pipelines, or produced by pipelines.
   "location": "us-central1"
 }
 ```
+
+```js
+// A S3 object
```
Comment (dorbin) [inline reference/artifacts/index.md]: s/A/An

File: reference/artifacts/types/s3-object/index.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+| `name` | Name/Path of the object. |
+| `location` | The region where your bucket was created |
+
+## Example
+
+```js
+{
+  "type": "s3/object",
+  "name": "s3://bucket/file.json",
+  "location": "us-east-1"
```
Comment (dorbin) [inline reference/artifacts/types/s3-object/index.md:29]: In the example you added to /reference/artifacts/index.md you included `reference` and `version` fields. Should those be added here, and to the list of fields above?

**Review by dorbin — APPROVED** [review]
Thanks for the doc, Benjamin!

File: reference/artifacts/types/s3-object/index.md (hunk 2 above)
Comment (benjaminws) [reply reference/artifacts/types/s3-object/index.md:29]: So it's actually the other way around. S3 doesn't version the files the same way, so those aren't used here. Updated in the example

File: reference/artifacts/index.md (hunk 1 above)
Comment (benjaminws) [reply reference/artifacts/index.md]: Fixed!

**Review by lwander — APPROVED** [review]
Thanks!

**Pull request merged**